src/shims/env.rs

```diff
@@ -321,7 +322,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             "`getcwd` is only available for the UNIX target family"
         );
 
-        this.check_no_isolation("`getcwd`")?;
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("getcwd", reject_with)?;
+            let err = Error::new(ErrorKind::NotFound, "rejected due to isolation");
```

Comment:
Ah, I forgot about one discussion we had: didn't we say this should be `PermissionDenied`? And some `match` for windows error conversions should be extended?

If you don't want to do this now, please leave FIXME in the code to change the error type.

Reply:
Also we should probably change `set_last_error_from_io_error` to take an error kind so that we don't have to make up a string here. But that, too, could be a separate PR if you prefer.

Sorry I forgot about these things in my last round of review.

Reply:
> so that we don't have to make up a string here.

I believe you could also use `Error::from(ErrorKind::...)` or `ErrorKind::....into()` to avoid the string:
https://doc.rust-lang.org/nightly/std/io/struct.Error.html#impl-From%3CErrorKind%3E

Reply:
But is there even any reason that `set_last_error_from_io_error` would take more than an `ErrorKind`?

Reply:
I specifically used `Error::new` as it allows custom payload for error not originated from OS. Quoting its documentation -- `This function is used to generically create I/O errors which do not originate from the OS itself.`. Shouldn't we use a custom message indicating that the problem is due to isolation so that user doesn't confuse it with an actual OS error? Or the intention is to make the distinction not clear?

As for which code to generate, yes, I mentioned using `PermissionDenied` earlier. But then, if it's a bogus error, does it matter?

For Windows code, I didn't want to mix it with this PR which is mainly about adding this new setup. I'll create a new PR for it. Also, we still have quite a few shims using old "always abort" setup, which I was planning to convert in next PR(s). Otherwise this new flag isn't very useful 🙂

Reply:
> But is there even any reason that `set_last_error_from_io_error` would take more than an `ErrorKind`?

No, I don't think so. I'm just pointing out an option if that function is not going to be changed as part of this PR.

> Shouldn't we use a custom message indicating that the problem is due to isolation so that user doesn't confuse it with an actual OS error?

The `set_last_error_from_io_error()` function only uses `e.kind()`, so I think the message will be discarded anyway.

Reply:
Ah, I didn't see that. Thanks for pointing out. I guess I can change it to take `ErrorKind`. @RalfJung Do you prefer I do it in this PR?

Reply:
> Do you prefer I do it in this PR?

Yeah I think that would be better. Thanks. :)

Reply:
Fixed the parameter in a new commit, which I didn't squash as it's independent from the first commit.

Reply:
Thanks a lot. :)

I think we probably want to change to a different error code, but we can do that later.